### PR TITLE
Find and locate range query item in the 'and' path of a query.

### PR DESCRIPTION
**Pull request by baldersheim**
If found and correlates with match phase limiter, copy the limits into the limiter.

@havardpe PR

### CODE REVIEW

File: searchcore/src/tests/proton/matching/match_phase_limiter/match_phase_limiter_test.cpp [hunk 1]
```diff
@@ -90,6 +100,15 @@ struct MockSearchable : Searchable {
     }
 };
 
+class MockRangeLocator : public RangeQueryLocator {
+public:
+    RangeLimitMetaInfo locate() const override {
+        return {};
+    }
+
+public:
```
Comment (havardpe) [inline searchcore/src/tests/proton/matching/match_phase_limiter/match_phase_limiter_test.cpp:109]: double public

File: searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp [hunk 2]
```diff
@@ -207,12 +209,17 @@ MatchToolsFactory(QueryLimiter               & queryLimiter,
         trace.addEvent(5, "Prepare shared state for multi-threaded rank executors");
         _rankSetup.prepareSharedState(_queryEnv, _queryEnv.getObjectStore());
         _diversityParams = extractDiversityParams(_rankSetup, rankProperties);
-        DegradationParams degradationParams = extractDegradationParams(_rankSetup, rankProperties);
+        vespalib::string attribute = DegradationAttribute::lookup(rankProperties, rankSetup.getDegradationAttribute());
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp:212]: be consistent with using rankSetup vs _rankSetup (member seems to be used elsewhere)

File: searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp [hunk 3]
```diff
@@ -207,12 +209,17 @@ MatchToolsFactory(QueryLimiter               & queryLimiter,
...
+        DegradationParams degradationParams = extractDegradationParams(_rankSetup, attribute, rankProperties);
+        const search::fef::FieldInfo * fieldInfo = indexEnv.getFieldByName(attribute);
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp:214]: no need to set up range term locator if degradation is disabled

File: searchcore/src/vespa/searchcore/proton/matching/rangequerylocator.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            for (size_t i(0); i < intermediate.childCnt(); i++) {
+                RangeLimitMetaInfo childMeta = locateFirst(field_id, intermediate.getChild(i));
+                if (childMeta.valid()) {
+                    return childMeta;
+                }
+            }
+        }
+    } else {
+        const Blueprint::State & state = blueprint.getState();
+        // TODO: Find fieldId
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/matching/rangequerylocator.cpp:44]: obsolete?

**Review by havardpe — APPROVED** [review]

File: searchcore/src/tests/proton/matching/match_phase_limiter/match_phase_limiter_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/matching/match_phase_limiter/match_phase_limiter_test.cpp:109]: Fixed

File: searchcore/src/vespa/searchcore/proton/matching/rangequerylocator.cpp (hunk 4 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/rangequerylocator.cpp:44]: Yes, fixed

File: searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp:212]: It was inconsistent multiple places. Unified on member

File: searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp (hunk 3 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp:214]: Good point, done

**Pull request merged**